Documentation/ceph-advanced-configuration.md

```diff
@@ -293,6 +294,83 @@ data:
     osd pool default size = 2
 ```
 
+## Custom CSI ceph.conf Settings
+
+> **WARNING**: It is highly recommended to use the default setting that comes with
+> CephCSI and this can only be used when absolutely necessary.
+> The `ceph.conf` be reset back to default values if/when the configurations are no
+> longer necessary.
+
+If the `csi-ceph-conf-override` ConfigMap is created before the cluster is
+started, the CephCSI pods
+will automatically pick up the settings. If you add the settings to the ConfigMap after the cluster
+has been initialized, you can restart the CSI pods to take immediate effect.
+
+After the CSI pods are restarted, the new settings should be in effect.
+
+### Example
+
+In this example we will set the `rbd_validate_pool` to `false` to skip rbd pool
+validation
+
+> **WARNING**: Modify Ceph settings carefully to avoid modifying the default
+> configuration.
+> Changing the settings could result in unexpected results if used incorrectly.
+
+```console
+kubectl -n rook-ceph get ConfigMap csi-ceph-conf-override -o yaml
+```
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: csi-ceph-conf-override
+  namespace: rook-ceph
+data:
+  ceph.conf: |
+    [global]
+    auth_cluster_required = cephx
+    auth_service_required = cephx
+    auth_client_required = cephx
+
+    fuse_set_user_groups = false
+    fuse_big_writes = true
+  # keyring is a required key and its value should be empty
+  keyring: |
+```
+
+To apply your desired configuration, you will need to update this ConfigMap. The next time the
+CephCSI pod(s) restarts, they will use the updated configs.
+
+```console
+kubectl -n rook-ceph edit configmap csi-ceph-conf-override
+```
+
+> **WARNING**: Don't delete/modify the keyring value it should be left empty.
+> Changing keyring settings could result in unexpected results.
+
+Modify the settings and save. Each line you add should be indented from the `ceph.conf` property as such:
+
+```yaml
```

Comment:
Can we keep this section simpler by having just the one example above? Showing a single line added could be done in the example above.